pkg/shell/terraform.go

```diff
@@ -136,9 +148,29 @@ func getOutputs(tf *tfexec.Terraform) (map[string]cty.Value, error) {
 		}
 	}
 
+	var apply bool
 	if wantsChange {
-		return map[string]cty.Value{},
-			fmt.Errorf("cloud infrastructure requires changes; please run \"terraform -chdir=%s apply\"", tf.WorkingDir())
+		log.Println("cloud infrastructure requires changes")
+		switch applyBehavior {
+		case AutomaticApply:
+			apply = true
+		case PromptBeforeApply:
+			apply = AskForConfirmation(fmt.Sprintf("Do you want to deploy group %s", tf.WorkingDir()))
```

Comment:
I think `no` should stop the process. If we want a skip function as was discussed that should be a third option.

Currently if I say not it continue to try to deploy the next groups:
```
2023/05/10 23:58:17 initializing terraform directory igc-01/one
2023/05/10 23:58:22 testing if terraform state of igc-01/one is in sync with cloud infrastructure
2023/05/10 23:58:24 cloud infrastructure requires changes
Do you want to deploy group igc-01/one [y/n]: n
2023/05/10 23:58:28 collecting terraform outputs from igc-01/one
2023/05/10 23:58:28 group one contains no artifacts to export
2023/05/10 23:58:28 collecting outputs for group two from group one
Error: consider running "ghpc export-outputs igc-01/one" (detailed error below)
The configuration file "igc-01/.ghpc/artifacts/one_outputs.tfvars" could not be read.
```

I think it would be reasonable to spit out the same exit as from export when changes are needed:

```
2023/05/11 00:01:12 testing if terraform state of igc-01/one is in sync with cloud infrastructure
2023/05/11 00:01:14 cloud infrastructure requires changes
Error: cloud infrastructure requires changes; please run "terraform -chdir=igc-01/one apply"
```

Reply:
I will alter this to have a 3d skip-and-continue option.

Reply:
Continuing this conversation at https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/1304#discussion_r1191575429 and resolving this conversation.